python/graphstorm/gconstruct/transform.py

```diff
@@ -577,7 +577,11 @@ def parse_feat_ops(confs):
         assert 'feature_col' in feat, \
                 "'feature_col' must be defined in a feature field."
         feat_name = feat['feature_name'] if 'feature_name' in feat else feat['feature_col']
-        out_dtype = _get_output_dtype(feat['out_dtype']) if 'out_dtype' in feat else None
+
+        # By default the out_dtype is set to float32
+        # FeatTransform sub-classes can ignore out_dtype
+        # or redefine it in its init function.
+        out_dtype = _get_output_dtype(feat['out_dtype']) if 'out_dtype' in feat else np.float32
```

Comment:
what about one-hot encoding? we decide to use int8 to store the data.

Reply:
The One-hot encoding transformer can ignore this out_dtype and has its own strategy.

Reply:
The conversion is done inside each transformation.

Reply:
CategoricalTransform just ignore out_dtype.